Pod/API/API.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  API.swift
+//  ElJayKit
+//
+//  Created by Isaac Greenspan on 12/10/15.
+//  Copyright © 2015 Isaac Greenspan. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+import AlamofireXMLRPC
+import MD5Digest
+
+/**
+ *  Provides LiveJournal API functionality for a given server and user.
+ */
+public struct API {
+    public typealias BaseCallback = Of<Any>.callback
+    
+    /// The client version to send to the server
+    public var clientVersion: ClientVersion! = ClientVersion() {
+        didSet {
+            if self.clientVersion == nil {
+                self.clientVersion = ClientVersion()
+            }
+        }
+    }
+    
+    /// The Alamofire Manager to use to make calls
+    private let manager: Manager
+    /// The target server
+    let server: Server
+    /// The user
+    let username: String
+    /// The MD5 hash of the password
+    private let passwordMD5: String
+    
+    /**
+     Create an API instance for the given server and the given user with the given plaintext password.
+     
+     - parameter server:   The server to target
+     - parameter username: The user
+     - parameter password: The user's plaintext password (only its MD5 hash is stored)
+     
+     - returns: The API instance
+     */
+    public init(server: Server, username: String, password: String, manager: Manager = Manager.sharedInstance) {
+        self.init(server: server, username: username, passwordMD5:(password as NSString).MD5Digest(), manager: manager)
+    }
+    
+    /**
+     Create an API instance for the given server and the given user with the given plaintext password.
+     
+     - parameter server:   The server to target
+     - parameter username: The user
+     - parameter passwordMD5: The MD5 hash of the user's password
+     
+     - returns: The API instance
+     */
+    public init(server: Server, username: String, passwordMD5: String, manager: Manager = Manager.sharedInstance) {
+        self.server = server
+        self.username = username
+        self.passwordMD5 = passwordMD5
+        self.manager = manager
+    }
+    
+    /**
+     Basic internal XMLRPC call wrapper.
+     
+     - parameter method:     The method to call
+     - parameter parameters: The parameters of the call, if any
+     - parameter callback:   The callback to execute with the result of the call
+     */
+    func simpleExecute(method: Method, parameters: [String: Any]? = nil, callback: BaseCallback) {
+        let paramArray: [Any]?
+        if let params = parameters {
+            paramArray = [params]
+        } else {
+            paramArray = nil
+        }
+        self.manager
+            .requestXMLRPC(
+                self.server.xmlrpcEndpoint,
+                methodName: method.methodName,
+                parameters: paramArray)
+            .responseXMLRPC({ response in
```

Comment:
[I'll be deep in the cold, cold ground before I ~~recognize Missourah~~](https://youtu.be/EM3NSTbi4OQ) use trailing closure syntax.

It's one of the things I really wish hadn't been brought in from Ruby.  It's nice if you're making a function that you want to look like a language construct (e.g., defining `unless (condition) { stuff }`), but for things that are clearly function/method calls, it feels like it only obscures what's going on.